Core/src/org/sleuthkit/autopsy/mainui/nodes/DataArtifactTypeFactory.java

```diff
@@ -60,33 +55,21 @@ public DataArtifactTypeFactory(Long dataSourceId) {
     }
 
     @Override
-    public boolean isRefreshRequired(PropertyChangeEvent evt) {
-        String eventType = evt.getPropertyName();
-        if (eventType.equals(IngestManager.IngestModuleEvent.DATA_ADDED.toString())) {
-            /**
-             * This is a stop gap measure until a different way of handling the
-             * closing of cases is worked out. Currently, remote events may be
-             * received for a case that is already closed.
-             */
-            try {
-                Case.getCurrentCaseThrows();
-                /**
-                 * Due to some unresolved issues with how cases are closed, it
-                 * is possible for the event to have a null oldValue if the
-                 * event is a remote event.
-                 */
-                final ModuleDataEvent event = (ModuleDataEvent) evt.getOldValue();
-                if (null != event && Category.DATA_ARTIFACT.equals(event.getBlackboardArtifactType().getCategory())
-                        && !(DataArtifactDAO.getIgnoredTreeTypes().contains(event.getBlackboardArtifactType()))) {
-                    return true;
-                }
-            } catch (NoCurrentCaseException notUsed) {
-                /**
-                 * Case is closed, do nothing.
-                 */
+    protected TreeItemDTO<DataArtifactSearchParam> getInvalidatedChild(TreeEvent daoEvt) {
+        if (daoEvt.getItemRecord().getTypeData() instanceof DataArtifactSearchParam) {
```

Comment:
We should rename getTypeData() to getSearchParam() if that is what we are going to store in there.